Build failing on macOS runners due to low default Gradle memory

The GitHub Actions build is failing only on **macOS runners**, while the same build passes successfully on **Linux and Windows.**

<details>
  <summary>Detailed Log here</summary>
  <p>
> Task :patchPluginXml
> Task :generateManifest
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :copyDeps
> Task :compileKotlin NO-SOURCE
> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
The message received from the daemon indicates that the daemon has disappeared.

Build request sent: Build{id=bb571e79-b5bc-44f8-b286-03d6b33e9453, currentDir=/Users/runner/work/liberty-tools-intellij/liberty-tools-intellij/liberty-tools-intellij}
Attempting to read last messages from the daemon log...
Daemon pid: 5226
  log file: /Users/runner/.gradle/daemon/8.7/daemon-5226.out.log
----- Last  20 lines from daemon log file - daemon-5226.out.log -----
2025-08-31T05:30:47.570+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2025-08-31T05:30:47.570+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] resetting idle timer
2025-08-31T05:30:47.570+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] daemon is running. Sleeping until state changes.
2025-08-31T05:30:47.571+0000 [INFO] [org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy] Daemon is about to start building Build{id=bb571e79-b5bc-44f8-b286-03d6b33e9453, currentDir=/Users/runner/work/liberty-tools-intellij/liberty-tools-intellij/liberty-tools-intellij}. Dispatching build started information...
2025-08-31T05:30:47.571+0000 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 24: dispatching org.gradle.launcher.daemon.protocol.BuildStarted@e3f99c57
2025-08-31T05:30:47.573+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment] Configuring env variables: [PATH, GITHUB_REPOSITORY_OWNER, OPENJ9_JAVA_COMMAND_LINE, REF_LTI_TAG, GITHUB_REPOSITORY_ID, CHROMEWEBDRIVER, GITHUB_RUN_NUMBER, ANDROID_HOME, NUNIT3_PATH, GITHUB_RETENTION_DAYS, GITHUB_WORKSPACE, GITHUB_API_URL, GITHUB_RUN_ATTEMPT, LSP4IJ_BRANCH, PWD, GITHUB_JOB, GITHUB_ACTIONS, DOTNET_MULTILEVEL_LOOKUP, RUNNER_ENVIRONMENT, HOMEBREW_CELLAR, ANDROID_NDK, GITHUB_ENV, GITHUB_EVENT_PATH, XPC_FLAGS, LC_ALL, GITHUB_REF_TYPE, GITHUB_ACTOR_ID, REF_LSP4IJ, LC_CTYPE, SHLVL, RUNNER_TOOL_CACHE, GITHUB_SERVER_URL, JAVA_HOME, GITHUB_REF, RUNNER_NAME, BOOTSTRAP_HASKELL_NONINTERACTIVE, HOMEBREW_PREFIX, NUNIT_BASE_PATH, GITHUB_OUTPUT, GITHUB_STEP_SUMMARY, GITHUB_REPOSITORY, GITHUB_ACTION_REF, HOMEBREW_REPOSITORY, RCT_NO_LAUNCH_PACKAGER, CI, RUNNER_ARCH, GITHUB_SHA, ImageOS, ImageVersion, USE_LOCAL_PLUGIN, RUNNER_TRACKING_ID, RUNNER_WORKSPACE, GITHUB_REF_PROTECTED, EDGEWEBDRIVER, GITHUB_WORKFLOW, HOMEBREW_NO_INSTALL_CLEANUP, GITHUB_BASE_REF, PIPX_HOME, GITHUB_WORKFLOW_SHA, GITHUB_PATH, ACTIONS_RUNNER_ACTION_ARCHIVE_CACHE, GITHUB_EVENT_NAME, LOGNAME, INFOPATH, GITHUB_TRIGGERING_ACTOR, SHELL, AGENT_TOOLSDIRECTORY, HOMEBREW_CASK_OPTS, ANDROID_NDK_HOME, ANDROID_NDK_ROOT, VCPKG_INSTALLATION_ROOT, GITHUB_GRAPHQL_URL, ENABLE_RUNNER_TRACING, TMPDIR, RUNNER_OS, JAVA_HOME_21_arm64, DOTNET_ROOT, BOOTSTRAP_HASKELL_INSTALL_NO_STACK_HOOK, GOROOT_1_22_ARM64, GITHUB_WORKFLOW_REF, GITHUB_ACTION, __CF_USER_TEXT_ENCODING, GITHUB_ACTOR, PIPX_BIN_DIR, VM_ASSETS, GECKOWEBDRIVER, GOROOT_1_23_ARM64, LANG, GITHUB_REPOSITORY_OWNER_ID, GOROOT_1_24_ARM64, XPC_SERVICE_NAME, GITHUB_STATE, GITHUB_REF_NAME, ANDROID_NDK_LATEST_HOME, GITHUB_ACTION_REPOSITORY, HOMEBREW_NO_AUTO_UPDATE, ANDROID_SDK_ROOT, GITHUB_RUN_ID, USER, SSH_AUTH_SOCK, JAVA_HOME_17_arm64, XCODE_16_DEVELOPER_DIR, JAVA_HOME_11_arm64, RUNNER_TEMP, GITHUB_HEAD_REF, HOME]
2025-08-31T05:30:47.577+0000 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 22: received class org.gradle.launcher.daemon.protocol.CloseInput
2025-08-31T05:30:47.578+0000 [DEBUG] [org.gradle.launcher.daemon.server.DefaultDaemonConnection] thread 22: Received IO message from client: org.gradle.launcher.daemon.protocol.CloseInput@e91edbfe
2025-08-31T05:30:47.579+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.LogToClient] About to start relaying all logs to the client via the connection.
2025-08-31T05:30:47.579+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogToClient] The client will now receive all logging from the daemon (pid: 5226). The daemon log file: /Users/runner/.gradle/daemon/8.7/daemon-5226.out.log
2025-08-31T05:30:47.582+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogAndCheckHealth] Starting build in new daemon [memory: 512 MiB]
2025-08-31T05:30:47.582+0000 [INFO] [org.gradle.launcher.daemon.server.exec.ForwardClientInput] Closing daemon's stdin at end of input.
2025-08-31T05:30:47.583+0000 [INFO] [org.gradle.launcher.daemon.server.exec.ForwardClientInput] The daemon will no longer process any standard input.
2025-08-31T05:30:47.583+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] The daemon has started executing the build.
2025-08-31T05:30:47.584+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=1fc352c8-7218-40a6-b18d-7303624b61c0,javaHome=/Users/runner/work/liberty-tools-intellij/liberty-tools-intellij/liberty-tools-intellij/test-tools/liberty-dev-tools/jdk-21.0.6+7/Contents/Home,daemonRegistryDir=/Users/runner/.gradle/daemon,pid=5226,idleTimeout=10800000,priority=NORMAL,applyInstrumentationAgent=true,daemonOpts=--add-opens=java.base/java.util=ALL-UNNAMED,--add-opens=java.base/java.lang=ALL-UNNAMED,--add-opens=java.base/java.lang.invoke=ALL-UNNAMED,--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED,--add-opens=java.base/java.nio.charset=ALL-UNNAMED,--add-opens=java.base/java.net=ALL-UNNAMED,--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED,-XX:MaxMetaspaceSize=384m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant]
[org.jetbrains.intellij.platform] IntelliJ Platform Gradle Plugin is outdated: 2.7.0. Update `org.jetbrains.intellij.platform` to: 2.7.2
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
----- End of the daemon log -----


FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)
</p>
</details>

--------
<details>
  <summary>Java heap space error</summary>
  <p>
java.lang.OutOfMemoryError: Java heap space
	at java.base/java.io.UnixFileSystem.canonicalize(UnixFileSystem.java:166)
	at java.base/java.io.File.getCanonicalPath(File.java:626)
	at java.base/java.io.File.getCanonicalFile(File.java:651)
	at org.gradle.internal.FileUtils.canonicalize(FileUtils.java:202)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:104)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:96)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:91)
	at org.gradle.cache.internal.OnDemandFileAccess.readFile(OnDemandFileAccess.java:41)
	at org.gradle.cache.internal.FileBackedObjectHolder.get(FileBackedObjectHolder.java:49)
	at org.gradle.cache.internal.FileIntegrityViolationSuppressingObjectHolderDecorator.get(FileIntegrityViolationSuppressingObjectHolderDecorator.java:34)
	at org.gradle.launcher.daemon.registry.PersistentDaemonRegistry.getAll(PersistentDaemonRegistry.java:71)
	at org.gradle.launcher.daemon.registry.PersistentDaemonRegistry.getDaemonsMatching(PersistentDaemonRegistry.java:116)
	at org.gradle.launcher.daemon.registry.PersistentDaemonRegistry.getIdle(PersistentDaemonRegistry.java:84)
	at org.gradle.launcher.daemon.server.CompatibleDaemonExpirationStrategy.checkExpiration(CompatibleDaemonExpirationStrategy.java:54)
	at org.gradle.launcher.daemon.server.expiry.AllDaemonExpirationStrategy.checkExpiration(AllDaemonExpirationStrategy.java:47)
	at org.gradle.launcher.daemon.server.expiry.AnyDaemonExpirationStrategy.checkExpiration(AnyDaemonExpirationStrategy.java:44)
	at org.gradle.launcher.daemon.server.MasterExpirationStrategy.checkExpiration(MasterExpirationStrategy.java:73)
	at org.gradle.launcher.daemon.server.Daemon$DaemonExpirationPeriodicCheck.run(Daemon.java:266)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:358)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1595)
Error: The operation was canceled.
</p>
</details>

-----------
From the logs, the Gradle daemon is starting with only 512 MB heap (-Xmx512m), which is insufficient for building the IntelliJ plugin (especially with IntelliJ 2025.2 and JCEF). On macOS runners, this leads to the daemon being killed due to Out Of Memory (OOM), while Linux/Windows complete without issues.

--- a/gradle.properties
+++ b/gradle.properties
@@ -19,5 +19,5 @@ lemminxVersion=0.26.1
 lclsLemminxVersion=2.4
 lclsVersion=2.4
 
-# Set maximum heap size to 2 GB and minimum to 1 GB
+# Increase Gradle memory by setting maximum heap size to 2 GB and minimum to 1 GB to prevent OutOfMemoryError
 org.gradle.jvmargs=-Xmx2g -Xms1g -Dfile.encoding=UTF-8

